pkg/proxy/iptables/proxier.go

```diff
@@ -1298,6 +1298,16 @@ func (proxier *Proxier) syncProxyRules() {
 		}
 	}
 
+	// Drop the packets in INVALID state, which would potentially cause
+	// unexpected connection reset.
+	// https://github.com/kubernetes/kubernetes/issues/74839
+	writeLine(proxier.filterRules,
+		"-A", string(kubeForwardChain),
```

Comment:
Wouldn't this need to be near the start of `syncProxyRules` rather than near the end? Many packets have already been matched and redirected by this point...

Reply:
Yes, it's the first rule in KUBE-FORWARD chain. 

```
Chain KUBE-FORWARD (1 references)
target     prot opt source               destination 
*** NEW RULE INSERTED HERE ***
ACCEPT     all  --  anywhere             anywhere             /* kubernetes forwarding rules */ mark match 0x4000/0x4000
ACCEPT     all  --  10.36.0.0/14         anywhere             /* kubernetes forwarding conntrack pod source rule */ ctstate RELATED,ESTABLISHED
ACCEPT     all  --  anywhere             10.36.0.0/14         /* kubernetes forwarding conntrack pod destination rule */ ctstate RELATED,ESTABLISHED

```

Reply:
oh, all the (relevant) stuff before that is in `natRules`, not `filterRules`. ok

Reply:
FYI Calico is doing similar things for disabling ConntrackInvalid https://github.com/projectcalico/felix/pull/1424 .

The Calico implementation also works for a cluster that runs IPVS as a load-balancer with iptables-mode of k8s-proxy .

Reply:
Ah, good to know. Thanks to bring it up @cmluciano . not sure if it should be an option here.. any reason not to enable that option?